core/src/main/java/org/apache/carbondata/core/scan/processor/AbstractDataBlockIterator.java

```diff
@@ -102,6 +107,17 @@ public AbstractDataBlockIterator(BlockExecutionInfo blockExecutionInfo, FileHold
     this.executorService = executorService;
     this.nextBlock = new AtomicBoolean(false);
     this.nextRead = new AtomicBoolean(false);
+    List<FileSystem.Statistics> allStatistics = FileSystem.getAllStatistics();
```

Comment:
Move this code AbstractDetailQueryResultIterator and pass statistics object from there to this class.